src/main/java/datawave/query/util/DateFrequencyValue.java

```diff
@@ -0,0 +1,359 @@
+            if (key == null) {
+                log.info("The key can't be null", new Exception());
+                keyValue = "1313";
+                ordinalDayOfYear = new OrdinalDayOfYear(keyValue);
+            }
+            keyValue = key;
+            if (keyValue != null) {
+                if (keyValue.length() >= 4)
+                    ordinalDayOfYear = new OrdinalDayOfYear(keyValue.substring(3));
+            } else {
+                ordinalDayOfYear = new OrdinalDayOfYear("1313");
+            }
+        }
+        
+        public byte[] getYear() {
+            int year = '\u0000';
+            try {
+                String yearStr = keyValue.substring(4);
+                year = Integer.parseUnsignedInt(yearStr);
+                
+            } catch (NumberFormatException numberFormatException) {
+                log.error("the year could not be extract from ", keyValue, numberFormatException);
+            }
+            return Base256Compression.numToBytes(year);
+        }
+        
+    }
+    
+    public class OrdinalDayOfYear {
+        private String mmDD;
+        
+        public OrdinalDayOfYear(String monthDay) {
+            mmDD = monthDay;
+        }
+        
+        public int getDateOrdinal() {
+            if (mmDD.equals("1313")) // dummy date when mmDD was bad
+                return BADORDINAL;
+            else
+                return calculateOrdinal();
+        }
+        
+        private int calculateOrdinal() {
+            // TODO implement
+            return 0;
+        }
+        
+    }
+    
+    /**
+     * This is a helper class that will compress the yyyyMMdd and the frequency date concatenated to it without a delimiter
+     */
+    public static class Base256Compression {
+        
+        private String content;
+        
+        public Base256Compression(String content) {
+            this.content = content;
+        }
+        
+        public static byte[] numToBytes(long num) {
+            if (num == 0) {
+                return new byte[] {(byte) '\u0000', (byte) '\u0000', (byte) '\u0000', (byte) '\u0000'};
+            } else if (num < 256) {
+                return new byte[] {(byte) '\u0000', (byte) '\u0000', (byte) '\u0000', (byte) (num)};
+            } else if (num < 65536) {
+                return new byte[] {(byte) '\u0000', (byte) '\u0000', (byte) (num >>> 8), (byte) num};
+            } else if (num < 16777216) {
+                return new byte[] {(byte) '\u0000', (byte) (num >>> 16), (byte) (num >>> 8), (byte) num};
+            } else { // up to 2,147,483,647
+                return new byte[] {(byte) (num >>> 24), (byte) (num >>> 16), (byte) (num >>> 8), (byte) num};
+            }
+        }
+        
+        public static int bytesToInteger(byte[] byteArray) {
+            int result = 0;
+            if (byteArray == null)
+                return 0;
+            if (byteArray.length > 4)
+                return 2_147_483_647;
```

Comment:
Integer.MAX_VALUE ?

Reply:
@ivakegg ,  I am refactoring OrdinalDayOfYear.calculateOrdinal().  Don't review it for structure yet.